### PR TITLE
Add split ratio to columns block

### PR DESCRIPTION
**Pull request by bosschaert**
Unit tests included.

Fixes: #5

Test URLs:
- Original: https://www.sunstar-foundation.org/en/
- Before: https://main--sunstar-foundation--hlxsites.hlx.live/en/
- After: https://issue-5--sunstar-foundation--hlxsites.hlx.live/en/

- [X] New variations introduced in this PR

Variation Name    | Documentation
------------- | -------------
 columns (split)  | [doc-link](https://main--sunstar-foundation--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/columns&index=19)

### CODE REVIEW

**Review by dnbute — COMMENTED** [review]
What about this use-case
https://issue-5--sunstar-foundation--hlxsites.hlx.page/drafts/dbute/flex-cols

Screenshot 2023-11-08 at 14 34 16


Do we maybe always want to add the bigger percentage to the text column, or do we say that this only works for 1-row columns? wdyt

File: test/blocks/columns/columns.plain.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+  <div>
+    <div class="columns split-80-20 native-image-sizes">
+      <div>
+        <div id="column-a" data-valign="middle">
```
Comment (sdmcraft) [inline test/blocks/columns/columns.plain.html:4]: Here a,b,x,y seem to be siblings. I thought a,b and x,y should be siblings with 80-20 rule being applied to each sibling pair.

File: blocks/columns/columns.js [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+    }
+  }
+
+  if (ratios.length === 0) {
+    return;
+  }
+
+  let pctIdx = 0;
+
+  for (let i = 0; i < block.children.length; i += 1) {
```
Comment (sdmcraft) [inline blocks/columns/columns.js:18]: Nit: Given its browser JS, we could use something like 
```
block.querySelectorAll(':scope > div').forEach((item) => { 
  item.querySelectorAll(':scope > div').forEach((child) => {
     ...foo...
  });
});
```

File: blocks/columns/columns.js [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+    if (block.children[i].localName === 'div') {
+      for (let j = 0; j < block.children[i].children.length; j += 1) {
+        if (block.children[i].children[j].localName === 'div') {
+          block.children[i].children[j].style.flexBasis = ratios[pctIdx];
+          pctIdx += 1;
+
+          if (pctIdx >= ratios.length) {
+            pctIdx = 0;
```
Comment (sdmcraft) [inline blocks/columns/columns.js:26]: What does this do? Do the percentage cyclically repeat?

**Review by sdmcraft — APPROVED** [review]

File: blocks/columns/columns.js (hunk 3 above)
Comment (bosschaert) [reply blocks/columns/columns.js:26]: The flex-basis needs to be applied to every cell in the columns 'table'. If there are multiple rows it starts the next row at 0 again.

**Comment by bosschaert** [comment]
@dnbute wrote:

> Do we maybe always want to add the bigger percentage to the text column, or do we say that this only works for 1-row columns? wdyt

Currently the setup it simple. It applies equally to all rows in the table. So there can definitely be more than 1 row, but the percentages are the same for col1 in each row, and col2 in each row and so on.

Maybe if we need the alternating behaviour you're mentioning somewhere we can revisit and see how to handle that then?

File: test/blocks/columns/columns.plain.html (hunk 1 above)
Comment (bosschaert) [reply test/blocks/columns/columns.plain.html:4]: Yes there was something incorrect in the test .html resource. I fixed that now, well spotted!

**Pull request merged**